guidance.md

```diff
@@ -1 +1,139 @@
+
+### Proposal 1 - Updated SDK versioning proposal
+
+In order to address [Issue 3](#-Issue-3---Client-library-specific-issues-with-the-earlier-azure-API-versioning-guidance) listed above, the updated Azure SDK client library proposal is to, instead of introducing a new major library version for a new API version, *add support for multiple service api versions in a single library*. By default, the latest service API version at the point of the release of the library is used, but a developer can opt in to earlier API versions.
+
+The exact mechanism by which a developer picks which API version to use varies by language, but it generally involves passing in a specific Api Version to use when constructing a client objet.
+
+Example from the [.NET Azure SDK design guidelines](https://azure.github.io/azure-sdk/dotnet_introduction.html#): 
+```C#
+public class ConfigurationClientOptions : ClientOptions {
+
+    public ConfigurationClientOptions(ServiceVersion version = ServiceVersion.V2019_05_09) {
+        if (version == default) 
+            throw new ArgumentException($"The service version {version} is not supported by this library.");
+        }
+    }
+
+    public enum ServiceVersion {
+        V2019_05_09 = 1,
+    }
+}
+```
+
+It is worth pointing out that the tolerance for changes that may break a small subset of customers is higher between versions of a client library than for a service api since a developer/customer has to make an explicit change to their application in order to reference a new version of a library. This allows a customer to properly change their application on their own schedule before deploying it into production.
+
+The proposal is not entirely without issues. The ability to support multiple API versions is dependent on services not regularly making large *contract or semantical breaking changes* since those types of changes would be hard to abstract away in a client library. However, this class of changes would be problematic in the previous guidance as well, since there is a strong correlation between changes that are difficult to abstract away in a client library that supports multiple API versions and changes that would cause friction in updating to a new version of a client library.
+
+### Process and terminology changes
+
+#### Update the terminology in guidance from "breaking" to "changes that require a new API version".
+
+> TODO: I'd love to get proposals of better names, please...
+
+Teams assume they understand what breaking means without reading our guidelines. And they often use the incorrect definition of the term.
+
+#### Establish a clear approval process for larger changes.
+
+Establishing a two-tiered process with both an escalation and approval paths for truly breaking changes helps team understand expectations. As part of reviewing the API changes, the Azure REST API review board can help determine if further approval is necessary.
```

Comment:
My fear (and Mark's) is that engineers simply can't think deeply about how their changes affect customers. The engineer's level practically doesn't even matter because they don't have familiarity with how their feature du jour is exposed via REST or via client libraries in many languages and what's available in those languages (side-by side- loading, method overloads, kwargs, property functions vs fields, optional parameters, signed vs unsigned ints, JS's 2^53 signed-only numbers, etc.). 

So, we're afraid that almost any change except for trivial changes that we should explicitly enumerate in this document could wreck havoc on the customer. Again Jumbo blob is an example where I'm sure engineers thought, its just a bigger number and not think at all about how get blocks will return non-deserializable values now or how just the blob's content length value is not deserializable now to a 32-bit (un)signed int. 

So, I think what MUST happen is that any truly trivial change that matches an explicit example in this document (we need to add more of them probably), can be implemented first and then the API review boards kick in to verify. Any change outside of the trivial set MUST be reviewed first and approved before implementation works begins. If it doesn't happen this way, then there is too much attachment to the implementation and teams feel like they MUST ship it now no matter what -- existing customers be damned.